### PR TITLE
Custom slider implementation

### PR DESCRIPTION
**Pull request by federicobadini**
We were using swiper.js to implement sliders in the codebase.
Since the bundle size impact was relevant even if we were using just a few of the functionalities it included, we made a custom slider implementation.

### QA

1. Visit Vercel preview
2. Verify that all 3 slider usages (product gallery in product page + press and quotes section in the store homepage) are still looking and working as before.

### CODE REVIEW

**Comment by deltuh-vee** [comment]
The /store page is 404ing for me on this branch.
image

**Comment by federicobadini** [comment]
> The /store page is 404ing for me on this branch.

In the `-prod` deployment the page is still disabled via feature flag.
The store homepage is accessible in the [react-commerce](https://react-commerce-git-new-slider-ifixit.vercel.app/store) deployment

**Comment by deltuh-vee** [comment]
dev_block ✊ on merge conflicts.

**Comment by erinemay** [comment]
## Product page:
### Desktop ✔️ 
- Forward/back buttons continue to work and show active/inactive state appropriately
- Dragging images continues to work: pulling image all the way, pulling image part of the way and releasing. 
- Clicking on the dots works
- Clicking on the thumbnails
- Product image zoom still works

### Mobile
- Forward/back buttons continue to work and show active/inactive state appropriately
- Clicking on the dots works

The response to touch controls on mobile has changed. It's more difficult to scroll through the images. I have to make a full swipe across most of the width of the page to get to the next image. In my screen recording, I do the same actions on this branch and main.

<details><summary>screen recording</summary>
Here's from my Pixel. It's harder to experience on browserstack, but it's still apparent there on the iPhone 13.

(main first, then on this branch)

https://user-images.githubusercontent.com/72166715/234932645-b3179b36-15a5-4168-8826-99d0bff202dd.mp4

</details>

## Store
### Desktop ✔️ 
Sliding looks and behaves the same as on master

### Mobile
I'm seeing the same increased difficulty to R/L scroll on this branch than master.

It also looks like it's misunderstanding up-down scroll for left right scroll. It's almost impossible to scroll up or down by dragging up/down in the new slider. 

<details><summary>screen recording</summary>
From my Pixel, but confirmed happening on iPhone on Browserstack.

https://user-images.githubusercontent.com/72166715/234935521-92666820-0217-4c9a-9e9a-dd4bc2daeeb7.mp4

</details>

dev_block 🧷 on the change to touch response on mobile

**Comment by federicobadini** [comment]
un_dev_block 👍 

With the latest changes the animation should finally result much smoother

File: frontend/templates/product/sections/ProductOverviewSection/ProductGallery.tsx [hunk 1]
```diff
@@ -1,22 +1,24 @@
-import { Box, Button, Circle, Flex, Img, Text, VStack } from '@chakra-ui/react';
+import {
+   Box,
+   Button,
+   Circle,
+   Flex,
+   Img,
+   Text,
+   Th,
```
Comment (ianrohde) [inline frontend/templates/product/sections/ProductOverviewSection/ProductGallery.tsx:8]:
```suggestion
```
Defined but unused

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻 This looks great! I appreciate the thought that went into the clear naming, made things easier to follow.

File: frontend/templates/product/sections/ProductOverviewSection/ProductGallery.tsx [hunk 2]
```diff
@@ -248,75 +199,69 @@ function useCurrentImageIndex(
...
-               transition="300ms all"
-            >
-               <FaIcon icon={faArrowRight} color="white" />
-            </Circle>
-         </Button>
-      </>
-   );
-};
+const PreviousButton = ({ disabled, onClick }: NavigationButtonProps) => (
+   <Button
```
Comment (ianrohde) [inline frontend/templates/product/sections/ProductOverviewSection/ProductGallery.tsx:208]: Any particular reason these aren't separated into a shared component? Most props are repeated, except for left/right specifics.

**Comment by danielbeardsley** [comment]
deploy_block :+1:  on deciding if we need to address the comments.

File: packages/ui/slider/index.tsx [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+   activeIndex,
+   slideCount,
+   slidesPerView,
+   drag,
+   spaceBetween,
+   slidesToKeepOnLeft,
+}: ComputeTranslationProps) {
+   const slideWidth = `(100% - ${
+      (slidesPerView - 1) * spaceBetween
+   }px) / ${slidesPerView}`;
```
[the remark this answers is not in the document]
Comment (federicobadini) [reply packages/ui/slider/index.tsx:319]: Here `calc` is not part of the resulting string while in the other factor it is prepended.
That's why it was kept separate

**Comment by federicobadini** [comment]
un_deploy_block 👍

**Comment by deltuh-vee** [comment]
Carry Over QA 🎬

**Pull request merged**